### PR TITLE
testutil/compose: align define with the rest

### PR DESCRIPTION
**Pull request by corverroos**
Aligns the `define` step with the other two steps (`lock` and `run`) by also creating a `docker-compose.yml` file as output.
This also ensures we test `charon create dkg` with compose. Just do noop for other keygens in this step.

category: refactor
ticket: #568

### CODE REVIEW

File: testutil/compose/config.go [hunk 1]
```diff
@@ -45,43 +42,53 @@ const (
...
-	// keyGen defines the key generation process.
-	KeyGen keyGen `json:"key_gen"`
+	// KeyGen defines the key generation process.
+	KeyGen KeyGen `json:"key_gen"`
 
 	// BeaconNode url endpoint or "mock" for simnet.
 	BeaconNode string `json:"beacon_node"`
-
-	// Def is the cluster definition.
-	Def cluster.Definition `json:"definition"`
```
Comment (corverroos) [inline testutil/compose/config.go:75]: cluster definition (and all other charon dependencies) completely removed. Only using docker-compose as API.

File: testutil/compose/config.go [hunk 2]
```diff
@@ -45,43 +42,52 @@ const (
...
 
+	// NumNodes is the number of charon nodes in the cluster.
+	NumNodes int `json:"num_nodes"`
+
+	// Threshold required for signature reconstruction. Defaults to safe value for number of nodes/peers.
+	Threshold int `json:"threshold"`
+
+	// NumValidators is the number of DVs (n*32ETH) to be created in the cluster lock file.
+	NumValidators int `json:"num_validators"`
+
```
Comment (corverroos) [inline testutil/compose/config.go:67]: added these since cluster definition was removed

File: testutil/compose/config.go [hunk 3]
```diff
@@ -45,43 +42,52 @@ const (
...
-// newBaseConfig returns a new base config excluding cluster definition.
-func newBaseConfig() config {
-	return config{
-		Version:    version,
-		ImageTag:   defaultImageTag,
-		VCs:        []vcType{vcTeku, vcLighthouse, vcMock},
-		KeyGen:     defaultKeyGen,
-		BeaconNode: defaultBeaconNode,
+// NewDefaultConfig returns a new default config.
+func NewDefaultConfig() Config {
```
Comment (corverroos) [inline testutil/compose/config.go:82]: export default config so flags can provide proper default values

**Review by xenowits — APPROVED** [review]
LGTM🚀

**Pull request merged**